app/upgrades/v3/upgrades_test.go

```diff
@@ -0,0 +1,82 @@
+package v3_test
+
+import (
+	"fmt"
+	"testing"
+
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	"github.com/stretchr/testify/suite"
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	"github.com/Stride-Labs/stride/v3/app/apptesting"
+)
+var (
+	airdropIdentifiers = []string{"stride", "gaia", "osmosis", "juno", "stars"}
+)
+const dummyUpgradeHeight = 5
+
+
+type UpgradeTestSuite struct {
+	apptesting.AppTestHelper
+}
+
+func (s *UpgradeTestSuite) SetupTest() {
+	s.Setup()
+}
+
+func TestKeeperTestSuite(t *testing.T) {
+	suite.Run(t, new(UpgradeTestSuite))
+}
+
+func (suite *UpgradeTestSuite) TestUpgrade() {
+	testCases := []struct {
+		msg         string
+		pre_update  func()
+		update      func()
+		post_update func()
+		expPass     bool
+	}{
+		{
+			"Test that upgrade does not panic",
+			func() {
+				// Create pool 1
+				suite.Setup()
+			},
+			func() {
+				// run upgrade
+				// TODO: Refactor this all into a helper fn
```

Comment:
May as well move this to bottom of `test_helpers.go` now instead of adding it as a TODO

```go
func (s *AppTestHelper) ConfirmUpgradeSucceededs(upgradeName string, upgradeHeight int) {
   contextBeforeUpgrade := s.Ctx().WithBlockHeight(upgradeHeight - 1)
   contextAtUpgrade := s.Ctx().WithBlockHeight(upgradeHeight)

   plan := upgradetypes.Plan{Name: upgradeName, Height: upgradeHeight}
   err := s.App.UpgradeKeeper.ScheduleUpgrade(contextBeforeUpgrade, plan)
   s.Require().NoError(err)

   plan, exists := s.App.UpgradeKeeper.GetUpgradePlan(contextBeforeUpgrade)
   s.Require().True(exists)

   s.Require().NotPanics(func() {
	beginBlockRequest := abci.RequestBeginBlock{}
	s.App.BeginBlocker(contextAtUpgrade, beginBlockRequest)
   })
}
```

and then you can add it to the v3 and v4 test
```go
suite.ConfirmUpgradeSucceededs("v3", 5)
suite.ConfirmUpgradeSucceededs("v4", 5)
```

Reply:
Thank Sam this look good